src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnnecessaryParenthesesCheck.java

```diff
@@ -388,10 +388,15 @@ private static boolean isExprSurrounded(DetailAST ast) {
      */
     private static boolean isLambdaSingleParameterSurrounded(DetailAST ast) {
         final DetailAST firstChild = ast.getFirstChild();
-        return firstChild.getType() == TokenTypes.LPAREN
-                && firstChild.getNextSibling().getChildCount(TokenTypes.PARAMETER_DEF) == 1
-                && firstChild.getNextSibling().getFirstChild().findFirstToken(TokenTypes.TYPE)
-                        .getChildCount() == 0;
+        boolean result = false;
+        if (firstChild.getType() == TokenTypes.LPAREN) {
+            final DetailAST parameters = firstChild.getNextSibling();
+            if (parameters.getChildCount(TokenTypes.PARAMETER_DEF) == 1
```

Comment:
@romani Does it make sense we add similar method with this parameter in addition to the other?

Reply:
please rephase question, I do not understand question.

Reply:
Should we add method `hasChildren(TokenType)`? We have a `getChildCount(TokenType)` just like we have a `getChildCount()`.

Reply:
I would rather not extend API that much, this is good candidate for utilities. We can move this to separate issue to analyze amount or usage first.

Reply:
@rnveach , please resolve this item if you are ok with my reply, if not, lets move it to separate issue